src/actions/pointerActions.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Callback higher-order function invoked by the dispatcher upon any action related to 
+ * pointer events. 
+ * 
+ * @param {*} model 
+ * @param {*} toolHandler
+ */
+const reactToPointerActions = (model, toolHandler) => {
+  return (action) => {
+    switch (action.type) {
+      /**
+       * Action raised upon a mouseDown event on the SVG overlay. 
+       * 
+       * Payload:
+       *  - x: The x co-ordinate of the event in viewport co-ordinates (out of 100)
+       *  - y: The y co-ordinate of the event in viewport co-ordinates (out of 100)
+       */
+      case 'PRESS':
+        toolHandler.handleAction(action);
+        break;
+
+      /**
+       * Action raised upon a mouseMove event on the SVG overlay. 
+       * 
+       * Payload:
+       *  - x: The x co-ordinate of the event in viewport co-ordinates (out of 100)
+       *  - y: The y co-ordinate of the event in viewport co-ordinates (out of 100)
+       */
+      case 'MOVE':
+        toolHandler.handleAction(action);
+        break;
+
+      /**
+       * Action raised upon a mouseUp event on the SVG overlay. 
+       * 
+       * Payload:
+       *  - x: The x co-ordinate of the event in viewport co-ordinates (out of 100)
+       *  - y: The y co-ordinate of the event in viewport co-ordinates (out of 100)
+       */
+      case 'RELEASE':
+        toolHandler.handleAction(action);
+        break;
+
+      /**
+       * Action raised upon a mouseLeave event on the SVG overlay. 
+       * 
+       * Payload:
+       *  - x: The x co-ordinate of the event in viewport co-ordinates (out of 100)
+       *  - y: The y co-ordinate of the event in viewport co-ordinates (out of 100)
+       */
+      case 'LEAVE_CANVAS':
+        toolHandler.handleAction(action);
+        break;
+
+      default: 
+        break;
```

Comment:
add a console warn or something in the default case so we know if something is slipping uncaught. Generally you could squish all this into a single statement but I assume this is done for commenting clarity

Reply:
The warn here will be constantly triggered. In the dispatcher system, actions are passed to all callbacks registered to the given dispatcher. So here, `PRESS` would be passed to `engineActions`, `viewerActions`, and `pointerActions`. Press handlers will be triggered for each of the callbacks that it is mentioned in. 

Instead, I can add safeties that ensure that each callback only accepts a certain list of actions, and actions are not duplicated across callbacks